### ISSUE TITLE
Document the default address pool for local networks

### ISSUE THREAD

**Issue by trapier** [docker/docker.github.io#8663]
### Problem description

The default pool for `docker swarm init --default-addr-pool` is covered in the swarm documentation:

> By default Docker Swarm uses a default address pool `10.0.0.0/8` for global scope (overlay) networks. Every network that does not have a subnet specified will have a subnet sequentially allocated from this pool. In some circumstances it may be desirable to use a different default IP address pool for networks.

Ref: https://docs.docker.com/engine/swarm/swarm-mode/#configuring-default-address-pools

The analogous default for local networks (`dockerd --default-address-pool`) is not currently represented in the documentation.   The defaults are as follows (including swarm `global` networks as well for reference):

| Type   | Default Size | Pool           |
| ------ | ------------ | -------------- |
| global | /24          | 10.0.0.0/8     |
| local  | /16          | 172.17.0.0/12  |
| local* | /20          | 192.168.0.0/16 |

\* Local networks are allocated from 172.17.0.0/12, and then 192.168.0.0/16 once 172.17.0.0/12 is exhausted.

### Problem location

Filing this in docker.github.io, but it might be best to address by adding a section to https://github.com/docker/cli/blob/master/docs/reference/commandline/dockerd.md.

### Project version(s) affected

`dockerd --default-address-pool` was released with server 18.06 (https://github.com/moby/moby/pull/36396).

### Suggestions for a fix

Note the "default" default address pool via a new section on https://github.com/docker/cli/blob/master/docs/reference/commandline/dockerd.md. Adding a section might also present the opportunity to document the CLI syntax:

> As [cli] flag (singular)
> 
> ```
> dockerd --default-address-pool base=10.10.0.0/16,size=24
> ```
> 
> In `daemon.json` (plural):
> ```
> {"default-address-pools":[{"base":"172.80.0.0/16","size":24},{"base":"172.90.0.0/16","size":24}]}
> ```

Ref: https://github.com/moby/moby/pull/36396#discussion_r184568118

**Comment by trapier** [comment]
Thanks to @bluikko for pointing out this gap in documentation! https://github.com/moby/moby/pull/29376#issuecomment-479374450

**Comment by ulidtko** [comment]
2.5 years past — and the chances to find the actual default values, **this table**, are much better on StackOverflow than on https://docs.docker.com :-1: 

Type | Default Size | Default Pool
-- | -- | --
global | /24 | 10.0.0.0/8
local | /16 | 172.17.0.0/12
local* | /20 | 192.168.0.0/16

**Comment by bluikko** [comment]
The persistent problem with Docker documentation: seems to be geared towards explaining basics of things for for beginners (or developers not familiar with networking/server administration) and very thin on actual details. Basically it's developers trying to be sysadmins and failing. Results of throwing away decades of system administration experience in the name of "devops".

**Comment by sgohl** [comment]
but what if you need more than 255 swarm-overlay networks? then a pool-size of `/8` wouldn't be enough, right?
sounds strange, but I've really got the need. Lets be a service-provider, needing to create a network per stack for a customer, having ~500 customers. The mask per network can easily be `/26` or even `/28`, but I can't figure out how to achieve more than 255 networks :-(

**Comment by bahag-schlachterk** [comment]
*warning this comment may contain sarcasm*
ah wait, there is a solution to shortage on IPv4-Addresses, its called IPv6 and was originally already specified back in 1998.
However as noted by @bluikko: DevOps is sometimes "developers playing Sysadmin and failing" and therefore with the new shiny world of docker and docker swarm we still rely on IPv4, NAT, Masquerading, Portforwarding and all the hell of IPv4 in use today.
See also: https://github.com/moby/moby/issues/32850
So I don't even dare to dream that docker swarm is going to support IPv6 by any (documented) means soon.
*end of sarcasm*

**Comment by ulidtko** [comment]
@docker-robot no, stop. That people aren't screaming about an issue, doesn't make the issue disappear.

/remove-lifecycle stale

**Comment by bahag-schlachterk** [comment]
@candlerb: Right! Thanks for spotting this one.
As far as my experience goes your first proposal is the right one for the documentation.

**Comment by ulidtko** [comment]
Hey @candlerb, you're right.

Just drilled into the source — found [this definition](https://github.com/moby/libnetwork/blob/64b7a4574d1426139437d20e81c0b6d391130ec8/ipamutils/utils.go#L18-L21):
```
	localScopeDefaultNetworks            = []*NetworkToSplit{
		{"172.17.0.0/16", 16}, {"172.18.0.0/16", 16}, {"172.19.0.0/16", 16},
		{"172.20.0.0/14", 16}, {"172.24.0.0/14", 16}, {"172.28.0.0/14", 16},
		{"192.168.0.0/16", 20}}
	globalScopeDefaultNetworks = []*NetworkToSplit{{"10.0.0.0/8", 24}}
```
Applies at least to docker v20.10.12 and likely many more (but probably not all) versions. Verify via the `docker/libnetwork` commit link in [vendor.conf](https://github.com/docker/docker/blob/v20.10.12/vendor.conf#L50).

I'm updating the table above. Thanks for noting!

**Comment by candlerb** [comment]
@ulidtko: the table you have provided [here](https://github.com/docker/docs/issues/8663#issuecomment-956438889) still has errors AFAICS.

* one row says "172.20.0.0/16 | 1 subnet" but clearly it's 172.20.0.0/14 (4 subnets) from the source code you quoted.
* the global pool, 10.0.0.0/8 with /24 subnets, would give you 65,536 /24-sized subnets (10.0.0.0/24 to 10.255.255.0/24), unless there's some other constraint that stops allocation beyond 10.0.255.0/24
